core/src/main/java/io/grpc/ChannelImpl.java

```diff
@@ -281,6 +291,7 @@ public void transportTerminated() {
         transportShutdown();
         transports.remove(transport);
         if (shutdown && transports.isEmpty()) {
+          releaseDeadlineCancellationExecutor();
```

Comment:
This is no longer needed. There is only one place when `shutdown` is set to `true`, and you are already covering it.